docs/user_documentation.rst

```diff
@@ -219,3 +291,46 @@ The GUI workflow is dead simple:
   is needed (such as confirming dependencies to install), DevAssistant will
   ask you, so don't go get your coffee just yet.
 - After all is done, get your coffee and enjoy.
+
+Where are the assistants located?
+---------------------------------
+
+You might ask, where are DAPs installed to? And you might already know the sort answer -
+to ``~/.devassistant``. However, the long answer is little bit more complicated.
+
+First of all, there are two things defined: ``DEVASSISTANT_HOME`` and ``DEVASSISTANT_PATH``.
+Be default, ``DEVASSISTANT_HOME`` is set to ``~/.devassistant`` and ``DEVASSISTANT_PATH`` contains
+the following:
+
+- ``~/.devassistant``
+- ``/usr/local/share/devassistant``
+- ``/usr/share/devassistant``
+
+All DAPs are installed to ``DEVASSISTANT_HOME``, however, you can run assistants from all the
+locations present in ``DEVASSISTANT_PATH``. If the assistant with the same name is present in more
+of them, locations on the beginning of the list have bigger priority.
+
+Sometimes the values of ``DEVASSISTANT_HOME`` and ``DEVASSISTANT_PATH`` are different. For example
+when installing DAPs as root, ``~/.devassistant`` is no longer there and
+``/usr/local/share/devassistant`` is used as ``DEVASSISTANT_HOME`` instead. So all the users can use
+assistants installed by root. You can also redefine the values entirely by using environment
+variables. The following example will install DAP ``python`` to ``.bah``::
+
+   $ DEVASSISTANT_HOME=.bah da pkg install python
+
+However you now cannot use assistants from ``python`` DAP unless you use that environment variable
+again.
+
+Manipulating DEVASSISTANT_PATH is very similar, but directories defined in that variable are
+used together with the default ones. If you want to avoid that, define DEVASSISTANT_NO_DEFAULT_PATH
+as well, but be sure to define DEVASSISTANT_HOME then, because this will unset the default value.
+
+The ``pkg`` command line action is prepared to work with multiple directories. 
+
+- ``install`` will always install to ``DEVASSISTANT_HOME``, however when the package is already installed in some other location from ``DEVASSISTANT_PATH``, it will not be installed again unless the ``--reinstall`` option is used.
+- ``remove``/``uninstall`` will only remove DAPs from ``DEVASSISTANT_HOME`` unless you use the ``--all-paths`` option. If you want to remove packages from directories where you don't have write access, use root access.
+- ``update`` will update DAPs in ``DEVASSISTANT_HOME`` and if it needs to update packages in different locations, it will install the newer version to ``DEVASSISTANT_HOME`` while keeping the old one untouched. You can use the ``--all-paths`` option as well, to update the packages across all paths, but you might need root access for that as well.
```

Comment:
``update`` updates DAPs in ``DEVASSISTANT_HOME``. It may happen that the transaction requires packages in other locations to be updated too. In that case, those packages will be left untouched, and newer versions will be installed in ``DEVASSISTANT_HOME``. You may override this behavior by using the ``--all-paths`` option, but you may need to perform that action as root if you don't have write access in all the locations.